cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/WorkflowHttpHandler.java

```diff
@@ -345,6 +346,34 @@ private WorkflowToken getWorkflowToken(String namespaceId, String appName, Strin
   }
 
   @GET
+  @Path("/apps/{app-id}/workflows/{workflow-id}/runs/{run-id}/nodestates")
```

Comment:
should this instead be ...``runs/{run-id}/nodes/state``, so if you wanted the state for a particular node, you can have ``runs/{run-id}/nodes/{node-name}/state``? 

I ask because there is a jira open to return all the nodes of a workflow, which would be ``runs/{run-id}/nodes``